drivers/scheduler/k8s/k8s.go

```diff
@@ -2848,7 +2848,17 @@ func (k *K8s) ScaleApplication(ctx *scheduler.Context, scaleFactorMap map[string
 			}
 			newScaleFactor := scaleFactorMap[obj.Name+DeploymentSuffix]
 			*dep.Spec.Replicas = newScaleFactor
-			if _, err := k8sOps.UpdateDeployment(dep); err != nil {
+
+			t := func() (interface{}, bool, error) {
+				dep, err := k8sOps.UpdateDeployment(dep)
```

Comment:
I don't think this will handle the resourceVersion change. It will keep failing even with the retries. You have to do a "get" on the deployment again and modify the new object which will have the latest resourceVersion.

Reply:
ah right, will change